File: backend/src/main/java/vn/rescue/core/domain/entities/ReliefItem.java
package vn.rescue.core.domain.entities;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "relief_items")
public class ReliefItem {
    @Id
    private String id;

    @Field("item_name")
    private String itemName;

    private String unit;

    private String description;
    
    @Field("image_url")
    private String imageUrl;
}

File: frontend/lib/models/relief_item.dart
class ReliefItem {
  final String? id;
  final String itemName;
  final String unit;
  final String description;
  final String? imageUrl;

  ReliefItem({
    this.id,
    required this.itemName,
    required this.unit,
    required this.description,
    this.imageUrl,
  });

  factory ReliefItem.fromJson(Map<String, dynamic> json) {
    return ReliefItem(
      id: json['id'],
      itemName: json['itemName'] ?? '',
      unit: json['unit'] ?? '',
      description: json['description'] ?? '',
      imageUrl: json['imageUrl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'itemName': itemName,
      'unit': unit,
      'description': description,
      'imageUrl': imageUrl,
    };
  }
}
